crates/stages/src/pipeline/mod.rs

```diff
@@ -463,6 +471,64 @@ where
             }
         }
     }
+
+    /// Prunes the stages according to their prune modes.
+    pub async fn prune(&mut self) -> Result<(), PipelineError> {
```

Comment:
this PR introduces the ability to prune in the pipeline, but doesn't actually invoke it, right?

Reply:
that's right